packages/plugins/Solana/src/state/Connection/connection.ts

```diff
@@ -95,22 +170,38 @@ class Connection implements BaseConnection {
         await Providers[options?.providerType ?? this.providerType].switchChain(options?.chainId)
     }
     getNativeToken(options?: SolanaWeb3ConnectionOptions): Promise<FungibleToken<ChainId, SchemaType>> {
-        throw new Error('Method not implemented.')
-    }
-    getNativeTokenBalance(options?: SolanaWeb3ConnectionOptions): Promise<string> {
-        throw new Error('Method not implemented.')
-    }
-    getFungibleTokenBalance(address: string, options?: SolanaWeb3ConnectionOptions): Promise<string> {
-        throw new Error('Method not implemented.')
+        const token = createNativeToken(options?.chainId ?? ChainId.Mainnet)
+        return Promise.resolve(token)
+    }
+    async getNativeTokenBalance(options?: SolanaWeb3ConnectionOptions): Promise<string> {
```

Comment:
Could you try to read the balance from the connection? Like how the `getBlockNumber` is implemented.